### PR TITLE
1527 python310 upgrade

### PR DESCRIPTION
**Pull request by edDocMe360**
Edits to update to python 3.10.13 for utils. 
Had to ditch OrderedSet in favor of simple sets and update tests. 
Updated CI to use appropriate python version
For ticket: https://app.zenhub.com/workspaces/va-notify-620d21369d810a00146ed9c8/issues/gh/department-of-veterans-affairs/notification-api/1527

Associated API PR
https://github.com/department-of-veterans-affairs/notification-api/pull/1584

Tested By:
1. Ensuring that unit tests pass. Had to edit tests (set vs. list). 
2. Regression Tests: https://github.com/department-of-veterans-affairs/notification-api-qa/actions/runs/7402769180

### CODE REVIEW

**Comment by edDocMe360** [comment]
Unit tests are passing. 
Added into requirements.txt on Dev and regression tests run. 
https://github.com/department-of-veterans-affairs/notification-api-qa/actions/runs/7402769180

**Review by cris-oddball — CHANGES_REQUESTED** [review]
Please add the info in the PR's comment into the PR's description so that we can easily see how this was tested.

**Review by cris-oddball — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@master
-    - name: docker://python:3.8-bullseye
-      uses: docker://python:3.8-bullseye
+    - name: docker://python:3.10.13
+      uses: docker://python:3.10.13
```
Comment (k-macmillan) [inline .github/workflows/ci.yaml:10]: Not sure why the previous team used bullseye, but we should probably keep it the same if we don't know why they used it. There's a [3.10-bullseye](https://github.com/docker-library/python/blob/936a2ed61e9930bfec0de767e78877e29d288322/3.10/bullseye/Dockerfile)

https://hub.docker.com/_/python/

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cris-oddball) [reply .github/workflows/ci.yaml:10]: I looked into this. Future versions in development, but not yet stable, are “Bullseye” and “Bookworm.” I don't think we should be using these images regardless of what the previous team did. That base Python image that Ed chose is based on the most recent stable image of Debian, which seems far safer to me than an unstable version.

Comment (k-macmillan) [reply .github/workflows/ci.yaml:10]: I can't tell why they ever used bullseye. Rescinding this requested change. The API uses alpine, so I could see narrowing it down to that but identifying which is beyond the scope of this work and can be determined at a later time. python3.10.13 works and we'll call that good.

**Review by cris-oddball — DISMISSED** [review]

**Review by cris-oddball — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**